### ISSUE TITLE
Allow fast failure of websocket connection

### ISSUE THREAD

**Issue by tresf** [qzind/tray#182]
In a controlled environment, the additional time added while the WebSocket attempts alternative ports and hosts may be time better spent alerting the user right away with a fast failure so that the user can start to fix the connection problem by starting/installing QZ Tray, etc.  Otherwise, the fallback logic can run for up to 20 seconds before failing.

This feature request is to allow a fast-failure of the websocket (e.g. on first port and on first host).

**Comment by bberenz** [comment]
Request looks to already be possible, just needed documented.
Verify with f67b79c1fd425fcefa7830a5312fb5a789adf307

**Comment by tresf** [comment]
API page has been refreshed: https://qz.io/api/qz.websocket#.connect

**Comment by klabarge** [comment]
This syntax allows me to succesfully connect to port 8182:

```js
qz.websocket.connect({ port: {insecure: [8182]} });
```

When I use the below, it fails. Why won't it explicitly connect to 8283 (and the rest of the insecure ports)?

```js
qz.websocket.connect({ port: {insecure: [8283]} });
```
![screen shot 2017-04-27 at 9 23 21 pm](https://cloud.githubusercontent.com/assets/12505463/25510652/cab301d2-2b8f-11e7-9075-f8528187165a.png)

**Comment by tresf** [comment]
Those are fallback ports so they're​very seldom used.  You can close QZ Tray, start a temp socket listener using e.g. Python on 8182 and then relaunch QZ Tray and the fallback ports will be available.  We only fallback in Java when the ports are used up.  JavaScript doesn't know any better and tries them all, hence the feature request.

**Comment by tresf** [comment]
@klabarge a quick way to force QZ Tray to start on 8283... 

1. Stop QZ Tray
2. `python -m SimpleHTTPServer 8182`
3. Start QZ Tray

**Comment by tresf** [comment]
So I tested the above scenario and it passes, however, in it's current state one should **always** supply both `secure` and `insecure` ports when being used.
* If both `secure` and `insecure` aren't overridden, the fast-fail won't happen when the site's being accessed by the opposite-specified protocol.
* Secondly, wiping out the insecure ports alone doesn't force the software to use secure ports when accessing from insecure URL, instead `ws://localhost:undefined` is what shows in the console logs.  I feel like this behavior could be improved.

**Comment by tresf** [comment]
Closed via consolidated commit bdf4c57.
